src/common/utils/http.ts

```diff
@@ -66,21 +67,36 @@ export async function httpGet<TBody extends JsonMap | string = string , TData =
             encoding: undefined,
             headers,
         },
-    ) as TRequestCoreOptionsRequiredUriUrl;
+    );
 
-    const promisifiedRequest = promisify<TRequestCoreOptionsRequiredUriUrl, request.Response>(request);
-    const response = await promisifiedRequest(requestOptions);
-
-    const result = {
-        isFailure: response.statusCode < 200 || response.statusCode >= 300,
-        isSuccess: response.statusCode >= 200 && response.statusCode < 300,
-        url,
-        responseUrl: response.url,
-        statusCode: response.statusCode,
-        body: response.body,
-        data: callback ? undefined : response.body,
-        contentType: response.caseless.get("Content-Type"),
-    };
+    const result = await (async (opt): Promise<IHttpGetResult<TBody, TData>> =>
```

Comment:
This "one-liner" code compression makes the source hard to read.
I would suggest preserving the previous coding style: `const promisifiedRequest: Promise<IHttpGetResult<TBody, TData>> = new Promise((resolve, reject) => { ... }); const response = await promisifiedRequest(requestOptions);`, and to call the original `request`  API directly with `requestOptions` (no intermediary `async` function to pass `opt` parameter).

What do you think?

Reply:
Is it very useful, this arrow function will not be never called

Reply:
> Is it very useful, this arrow function will never be called

I am not sure I fully understand your reply, so let me try to explain what I mean by "hard to read" code:

Current:
```
const result = await(async (opt): Promise<IHttpGetResult<TBody, TData>> =>
    new Promise((resolve, reject) => {
        request(opt, (err, response) => {
            // ...
        });
    }))(requestOptions);
```

Proposed alternative:
```
const result: IHttpGetResult<TBody, TData> = await new Promise((resolve, reject) => {
    request(requestOptions, (err, response) => {
        // ...
    });
});
```

What do you think?

Reply:
ok